derp/derp_server.go

```diff
@@ -1383,7 +1383,14 @@ func (s *Server) verifyClient(ctx context.Context, clientKey key.NodePublic, inf
 
 	// tailscaled-based verification:
 	if s.verifyClientsLocalTailscaled {
-		_, err := s.localClient.WhoIsNodeKey(ctx, clientKey)
+		status, err := s.localClient.StatusWithoutPeers(ctx)
+		if err != nil {
+			return fmt.Errorf("can not get status: %w", err)
+		}
+		if status.Self.PublicKey == clientKey {
+			return nil
+		}
+		_, err = s.localClient.WhoIsNodeKey(ctx, clientKey)
```

Comment:
Can we just fix the server (tailscaled) side of WhoIsNodeKey to include self?

Reply:
Actually it looks like the server already handles this.

Oh, because it was just fixed in 99aaa6e92cda572896503538e1716851358e42d6 (#16096) for dup issue #16052.